src/Simulation/CsharpGeneration/Resources/EntryPointDriver.cs

```diff
@@ -0,0 +1,355 @@
+                return arg;
+            }
+        }
+
+        /// <summary>
+        /// The argument handler for the Q# BigInt type.
+        /// </summary>
+        internal static Argument<BigInteger> BigIntArgumentHandler => new Argument<BigInteger>(
+            CreateArgumentParser(value =>
+                BigInteger.TryParse(value, out var result) ? (true, result) : (false, default)));
+
+        /// <summary>
+        /// The argument handler for the Q# Range type.
+        /// </summary>
+        internal static Argument<QRange> RangeArgumentHandler => new Argument<QRange>(result =>
+        {
+            var option = ((OptionResult)result.Parent).Token.Value;
+            var value = string.Join(' ', result.Tokens.Select(token => token.Value));
+            return new[]
+            {
+                ParseRangeFromEnumerable(option, value, result.Tokens.Select(token => token.Value)),
+                ParseRangeFromEnumerable(option, value, value.Split(".."))
+            }
+            .Choose(errors => result.ErrorMessage = string.Join('\n', errors.Distinct()));
+        })
+        {
+            Arity = ArgumentArity.OneOrMore
+        };
+
+        /// <summary>
+        /// Runs the entry point.
+        /// </summary>
+        /// <param name="args">The command-line arguments.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> Main(string[] args)
+        {
+            var simulate = new Command("simulate", "(default) Run the program using a local simulator.");
+            // TODO: Replace the option aliases with constants defined in Microsoft.Quantum.QsCompiler.ReservedKeywords.
+            TryCreateOption(new[] { "--simulator", "-s" },
+                            () => @EntryPointAdapter.DefaultSimulator,
+                            "The name of the simulator to use.")
+                .Then(simulate.AddOption);
+            simulate.Handler = CommandHandler.Create<@EntryPointAdapter, string>(Simulate);
+
+            var root = new RootCommand(@EntryPointAdapter.Summary) { simulate };
+            foreach (var option in @EntryPointAdapter.Options) { root.AddGlobalOption(option); }
+
+            // Set the simulate command as the default.
+            foreach (var option in simulate.Options) { root.AddOption(option); }
+            root.Handler = simulate.Handler;
+
+            return await root.InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Simulates the entry point.
+        /// </summary>
+        /// <param name="entryPoint">The entry point adapter.</param>
+        /// <param name="simulator">The simulator to use.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> Simulate(@EntryPointAdapter entryPoint, string simulator)
+        {
+            switch (simulator)
+            {
+                case "ResourcesEstimator":
```

Comment:
Instead of strings, I suggest to use the AssemblyConstants.ResourcesEstimator etc.